### DISCUSSION TITLE
Is there a way to let a custom completer re-write the current line?

### DISCUSSION THREAD

**Discussion by dedebenui** [nushell/nushell#10615] [Q&A]
I set up my `zoxide` completer and it works great when I input only one word:
```
~ : z foo<tab>
~ : z /home/myself/foo<enter>
~/foo :
```

however, it doesn't work when the input is more than one word
```
~ : z foo ba<tab>
~ : z foo /home/myself/foo/bar<enter>
zoxide: no match found
```

Is there a way to let the custom completer rewrite the whole line? like this

```
~ : z foo ba<tab>
~ : z /home/myself/foo/bar<enter>
~/foo/bar :
```

Another solution would be to let the completer call a command upon accepting a suggestion:

```
~ : z foo ba<tab>
[/home/myself/foo/bar]    /mnt/drive/foo/baz
<enter>
~/foo/bar :
```

Is there any such thing in Nushell?

**Comment by fdncred** [comment]
The only way that I know to rewrite the repl command line is to use a keybinding or `commandline` command.
